_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp

```diff
@@ -378,12 +378,12 @@ namespace UMC_AV1_DECODER
                 updated_refs = prevFrameDPB;
                 Repeat_show = 0;
             }
-            Curr_temp = Curr;
+            Curr_temp.reset(Curr.get());
```

Comment:
This does not make sense to me. Unique_ptr is specifically designed to have single owner of the object. By assigning result of get() you actually end up with 2 unique_ptr managing the same object. It seems you need to have :
1. Either assignment as you had before so Curr will lose ownership
2. Or (if you still need ownership for Curr) you need you declare these as std::shared_ptr objects to have multiple owners.

Reply:
+1

Reply:
Same with my previous comments.